docs: document some learnings about mode (#124)
podium-lib/podlet-server: README.md

### Server Side Rendered (hydrate)

Use this mode if you're writing podlets that will be included in layouts you don't control.

When `app.mode` is set to `hydrate`, the content and fallback routes will be server rendered and then client side hydrated afterward.
No additional setup is required but see guidelines for writing SSR custom elements on the Lit docs site. [https://lit.dev/docs/ssr/authoring/](https://lit.dev/docs/ssr/authoring/).

### Server Side Rendered (ssr-only)

When `app.mode` is set to `ssr-only`, the content and fallback routes will be server rendered but will not be hydrated on the client side afterward.
This is good if you want to use components from a shared library or design system but don't actually need interactivity on the client side.

> If you control the layout(s) where the podlet will be rendered, it is recommended you use this option whenever possible, since it will save many KBs of JavaScript being sent to the users browser.

Reviewer: ```suggestion
If you control the layout(s) where the podlet will be rendered, it is recommended you use this option, as it will prevent sending KBs of JavaScript to the users browser.
```

> The downside is, if another part of a layout does client-side render a shared component (such as an icon), Lit won't be able to properly replace the server-side rendered markup, and you end up with duplicates.

Reviewer: The downside to what? 🤔  To using the recommendation in the line above? Or not using it? I was a bit confused reading this.

Reviewer 2: ooooh, so if I understand correctly, the situation is, you have 1 podlet that is mode: "csr-only" and one part that is "hydrate" and they both use the same icon?

And when you have this situation, you get a hydration mismatch (and therefore duplication) of the icon?

Author: `ssr-only` + essentially `csr-only` in this case (the other party isn't a `podlet-server`)

Author: ```suggestion
The downside compared to `hydrate` is, if another part of a layout does client-side render a shared component (such as an icon), Lit won't be able to properly replace the server-side rendered markup, and you end up with duplicates.
```
